crates/uv/tests/pip_sync.rs

```diff
@@ -2993,3 +2993,51 @@ requires-python = "<=3.5"
 
     Ok(())
 }
+
+/// Raise an error when a direct URL dependency's `Requires-Python` constraint is not met.
+///
+/// TODO(charlie): This currently passes, but should fail.
```

Comment:
Why does this one still pass?

Reply:
I honestly don't remember, let me try rebasing.

Reply:
Oh, I think we don't enforce this in `pip-sync`.

Reply:
I can't remember why. I think the separation somewhere in the pipeline made it hard.

Reply:
Should we track this somewhere?

Reply:
#2443